### PR TITLE
LPD-15212 Check the workflow XML file for malicious content

### PR DESCRIPTION
**Pull request by vngoliveira**
[LPD-15212](https://liferay.atlassian.net/browse/LPD-15212)
Hi team!
I worked on this fix to block any malicious content in the XML file. I chose this way because the validateDefinition function appeared to be the better place to do this verification. 
I thought about doing a replacement to sanitize the XML file in SourceBuilder.js, but after some trials, I saw it would be complex and leave the door open to more issues in the future.
Thanks for your attention and review. Feel free to contact me and ask about the fix!

### CODE REVIEW

**Comment by vngoliveira** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
Hey @guilhermedcamacho,
Could you assign someone to review this PR? :smile: 
Thank you!

**Comment by danvb112** [comment]
After talking with @vngoliveira, I'm wating she fix somethings in the PR

**Comment by vngoliveira** [comment]
ci:test:sf

**Comment by vngoliveira** [comment]
I fixed the PR, @danvb112

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/source-builder/xmlUtil.js [hunk 1]
```diff
@@ -165,8 +169,26 @@ const XMLUtil = {
...
+
+		let maliciousContentValidation;
+
+		for (const item of maliciousContent) {
+			const regex = new RegExp(item);
+			maliciousContentValidation = regex.test(definition);
+			if (maliciousContentValidation) {
+				break;
+			}
+		}
```
Comment (danvb112) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/source-builder/xmlUtil.js:187]: Hey @vngoliveira, you can use the `some` method from `javascript` here:

```tsx
const maliciousContentValidation = maliciousContent.some((item) =>
	new RegExp(item).test(definition)
);
```

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/source-builder/SourceBuilder.js [hunk 2]
```diff
@@ -53,6 +54,20 @@ export default function SourceBuilder() {
...
+					codeEditor.addEventListener('keyup', () => {
+						if (currentEditor.getData() !== xmlContent) {
+							const newXmlContent = currentEditor.getData();
+							const sanitizedXmlContent = newXmlContent.replace(
+								REGEX_ALERT,
+								''
+							);
+							currentEditor.setData(sanitizedXmlContent);
+						}
+					});
```
Comment (danvb112) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/source-builder/SourceBuilder.js:70]: @vngoliveira can you please add a blank line bellow?

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/source-builder/SourceBuilder.js [hunk 3]
```diff
@@ -53,6 +54,20 @@ export default function SourceBuilder() {
...
+					);
+
+					codeEditor.addEventListener('keyup', () => {
+						if (currentEditor.getData() !== xmlContent) {
+							const newXmlContent = currentEditor.getData();
+							const sanitizedXmlContent = newXmlContent.replace(
+								REGEX_ALERT,
+								''
+							);
+							currentEditor.setData(sanitizedXmlContent);
```
Comment (danvb112) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/source-builder/SourceBuilder.js:68]: @vngoliveira can you please add a blank line here please

**Review by danvb112 — CHANGES_REQUESTED** [review]
Het @vngoliveira, good work, I have some suggestions :smile:

**Comment by vngoliveira** [comment]
ci:test:relevant

**Review by danvb112 — APPROVED** [review]
LGTM! :tada:

**Comment by danvb112** [comment]
ci:test:workflow

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:report:2533573076

**Comment by marcelabc** [comment]
ci:report:2534136265

**Comment by marcelabc** [comment]
I analyzed and there are no unique failures :+1

**Comment by marcelabc** [comment]
ci:forward:force

**Pull request closed without merging**